bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -130,8 +128,15 @@ public MavenWorkspaceRepository() {
 
 	@Activate
 	void activate() {
-		process();
-		list = memoize(this::list0);
+		try {
+			Central.bndCall(() -> {
+				process();
+				list = memoize(this::list0);
```

Comment:
This does not do what you want. It just creates the memoize object. You need to wrap the supplier. So move the bndCall invocation into list0.

Reply:
changed the logic